src/Illuminate/Queue/CallQueuedHandler.php

```diff
@@ -125,6 +127,28 @@ protected function dispatchThroughMiddleware(Job $job, $command)
                 });
     }
 
+    /**
+     * Dispatch the given job / command through its specified middleware.
+     *
+     * @param  \Illuminate\Contracts\Queue\Job  $job
+     * @param  mixed  $command
+     * @return mixed
+     */
+    protected function terminateThroughMiddleware(Job $job, $command)
+    {
+        if ($command instanceof \__PHP_Incomplete_Class) {
+            throw new Exception('Job is incomplete class: '.json_encode($command));
+        }
+
+        return (new Pipeline($this->container))->send($command)
+                                               ->through($command->middleware ?? [])
```

Comment:
I don't know if i need to recalculate the middleware array, i assume that it's already using the array merge from the `dispatchThroughMiddleware` method